### PR TITLE
[tools] Add option to print annotations in annotation checker

### PR DESCRIPTION
**Pull request by vpranckaitis**
**What this PR does / why we need it**:

Adds option to `annotation_checker` tool to print annotations of filtered series. Also, some CLI options and their descriptions are updated.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/cmd/tools/annotation_checker/main/README.md [hunk 1]
```diff
@@ -2,26 +2,30 @@
...
-                    Path prefix [e.g. /var/lib/m3db]
- -R, --annotation-rewritten
-                    Filters metrics with annotation rewrites
- -s, --shard=value  Shard number, or -1 for all shards in the directory
- -t, --fileset-type=value
-                    flush|snapshot
- -v, --volume=value
-                    Volume number
+Usage: annotation_checker [-P] [-a no-filtering|no-initial-annotation|annotation-rewritten] [-b nsec] [-f string] [-n string] [-p string] [-s int] [-t flush|snapshot] [-v int] [parameters ...]
+ -a, --annotation-filter=no-filtering|no-initial-annotation|annotation-rewritten
```
Comment (linasm) [inline src/cmd/tools/annotation_checker/main/README.md:12]: Nit: I suggest shortening things a little, e.g. ` --annotation-filter=no-filtering` -> ` --annotation-filter=none`,
also in `annotation_checker --annotation-filter=annotation-rewritten` we repeat `annotation` all over the place, perhaps could use `--annotation-filter=rewritten`.

File: src/cmd/tools/annotation_checker/main/main.go [hunk 2]
```diff
@@ -195,6 +229,38 @@ func checkAnnotations(data []byte, encodingOpts encoding.Options) (bool, bool, e
...
+	idx := 0
+	hasAnnotation := false
+	for iter.Next() {
+		_, _, annotationBytes := iter.Current()
+
+		if annotationBytes != nil {
+			hasAnnotation = true
+			payload.Reset()
+			err := payload.Unmarshal(annotationBytes)
+			if err != nil {
```
Comment (linasm) [inline src/cmd/tools/annotation_checker/main/main.go:246]: Nit:
```suggestion
			if err := payload.Unmarshal(annotationBytes); err != nil {
```

**Review by linasm — COMMENTED** [review]
This tool is becoming complicated. `read_data_files` is already printing annotations, is that implementation not sufficient (we could add protobuf unmarshaling there, currently it outputs base64)?

**Review by vpranckaitis — COMMENTED** [review]
> This tool is becoming complicated. `read_data_files` is already printing annotations, is that implementation not sufficient (we could add protobuf unmarshaling there, currently it outputs base64)?

I think it's way more convenient to have annotation printing here rather than in `read_data_files`:
- `read_data_files` prints all the datapoints, even if they don't have annotations. This tools ignores the datapoints without annotations;
- It would be way less convenient if you had to use `annotation_checker` to figure out which time series have missing / rewritten annotations, and then go to `read_data_files` to look at the annotations itself. Especially if you had to do this for multiple series (i.e. no single ID filter would cover all the time series you're interested in).

**Comment by linasm** [comment]
> > This tool is becoming complicated. `read_data_files` is already printing annotations, is that implementation not sufficient (we could add protobuf unmarshaling there, currently it outputs base64)?
> 
> I think it's way more convenient to have annotation printing here rather than in `read_data_files`:
> 
> * `read_data_files` prints all the datapoints, even if they don't have annotations. This tools ignores the datapoints without annotations;
> * It would be way less convenient if you had to use `annotation_checker` to figure out which time series have missing / rewritten annotations, and then go to `read_data_files` to look at the annotations itself. Especially if you had to do this for multiple series (i.e. no single ID filter would cover all the time series you're interested in).

But then, why add filtering by `id`? We separated this tool from `read_data_files` intentionally, but now it looks like they are converging in functionality. You can always filter by `id` in `read_data_files` and then grep the output for `annotation:` to filter out the noise.

**Review by Antanukas — APPROVED** [review]
LGTM. I guess read data files can be more of a raw data reader, and this tools is more focused on annotation related stuff.

**Pull request closed without merging**